server/cla.go

```diff
@@ -54,80 +59,108 @@ func (s *Server) checkCLA(pr *model.PullRequest) {
-				mlog.Error("Unable to create the github status for for PR", mlog.Int("pr", pr.Number), mlog.Err(errStatus))
-				return
-			}
-			mlog.Info("will clean some comments regarding the CLA")
-			commentToRemove, existComment := checkCLAComment(comments, s.Config.Username)
-			if existComment {
-				mlog.Info("Removing old comment with ID", mlog.Int64("ID", commentToRemove))
-				_, err := s.GithubClient.Issues.DeleteComment(context.Background(), pr.RepoOwner, pr.RepoName, commentToRemove)
-				if err != nil {
-					mlog.Error("Unable to remove old Mattermod comment", mlog.Err(err))
-				}
-			}
-			return
+func isNameInCLAList(usersWhoSignedCLA []string, author string) bool {
+	authorLowerCase := strings.ToLower(author)
+	for _, userToTrim := range usersWhoSignedCLA {
+		user := strings.TrimSpace(userToTrim)
+		if strings.Compare(user, author) == 0 || strings.Compare(user, authorLowerCase) == 0 || strings.Compare(strings.ToLower(user), authorLowerCase) == 0 {
+			return true
 		}
 	}
-
-	_, existComment := checkCLAComment(comments, s.Config.Username)
-	if !existComment {
-		s.sendGitHubComment(pr.RepoOwner, pr.RepoName, pr.Number, strings.Replace(s.Config.NeedsToSignCLAMessage, "USERNAME", "@"+username, 1))
-	}
-	claStatus.State = github.String(stateError)
-	userMsg := fmt.Sprintf("%s needs to sign the CLA", username)
-	claStatus.Description = github.String(userMsg)
-	mlog.Info("will post error on CLA", mlog.String("user", username))
-	_, _, errStatus := s.GithubClient.Repositories.CreateStatus(context.Background(), pr.RepoOwner, pr.RepoName, pr.Sha, claStatus)
-	if errStatus != nil {
-		mlog.Error("Unable to create the github status for for PR", mlog.Int("pr", pr.Number), mlog.Err(errStatus))
-		return
-	}
+	return false
 }
 
-func checkCLAComment(comments []*github.IssueComment, username string) (int64, bool) {
+func findNeedsToSignCLAComment(comments []*github.IssueComment, username string) (id int64, found bool) {
 	for _, comment := range comments {
-		if *comment.User.Login == username {
-			if strings.Contains(*comment.Body, "Please help complete the Mattermost") {
-				return *comment.ID, true
-			}
+		if *comment.User.Login == username && strings.Contains(*comment.Body, "Please help complete the Mattermost") {
+			return *comment.ID, true
 		}
 	}
 	return 0, false
 }
+
+func (s *Server) createCLAPendingStatus(pr *model.PullRequest) {
+	status := &github.RepoStatus{
+		State:       github.String(statePending),
+		Description: github.String("Checking if " + pr.Username + " signed CLA"),
+		TargetURL:   github.String(s.Config.SignedCLAURL),
+		Context:     github.String(s.Config.CLAGithubStatusContext),
+	}
+	ctx, cancel := context.WithTimeout(context.Background(), requestTimeoutGithub)
+	defer cancel()
+	err := s.createRepoStatus(ctx, pr, status)
+	if err != nil {
+		s.logToMattermost("failed to create status for PR: " + strconv.Itoa(pr.Number) + " Context: " + s.Config.CLAGithubStatusContext + " Error: ```" + err.Error() + "```")
```

Comment:
Drive by idea: Maybe all error of a certain log level should to to Mattermost, e.g. `Error`

Reply:
Out of scope, but yeah this is a long term goal of mine. :+1: